### DISCUSSION TITLE
nuxt 3 dynamic routing, problems

### DISCUSSION THREAD

**Discussion by Mamush-Meshesha** [nuxt/nuxt#28039] [Questions]
my folder structure is like this,
| pages/
---| about.vue
---| catagory.vue
---| catagory/
-----| [name].vue

in catagory.vue file i have nuxtlink for 
```
 <div class="flex justify-center" v-for="recipe in result.recipe" :key="recipe.id">
          <NuxtLink :to="{ name: 'catagory-name', params: { name: recipe.catagory.name } }">
            <div v-for="image in recipe.images" :key="image.id">
              <img
                :src="image.url"
                alt="image not found"
                class="md:w-56 w-44 md:h-56 h-44 rounded-full object-cover"
              />
              <h1 class="">{{ recipe.catagory.name }}</h1>
            </div>
          </NuxtLink>
        </div>
        ```
        and in [name].vue file, i have this
        
 <script setup>
            const route = useRoute();
            const name = route.params.catagory.name
<script>

the error is 
missing required params name

**Comment by xjccc** [comment]
`route.params.name`?

**Reply by Mamush-Meshesha** [reply]
i used it like this , still the same
route.params.name

**Reply by xjccc** [reply]
emm... Y could change your `catagory.vue` -> `catagory/index.vue`

[example](https://stackblitz.com/edit/nuxt-starter-hpbksw?file=app.vue,pages%2Fcate%2F[name].vue,pages%2Findex.vue,pages%2Fcate%2Findex.vue)
